base/array.jl

```diff
@@ -1987,3 +1987,139 @@ julia> symdiff([1,2,3],[3,4,5],[4,5,6])
 ```
 """
 symdiff(a, b, rest...) = symdiff(a, symdiff(b, rest...))
+
+## replace/replace! ##
+
+# NOTE: to implement replace! and replace for a new type T, it's enough to define:
```

Comment:
I don't think this is the place to document this. Though I don't really know where it could go. Also, I don't like recommending people to override an unexported function whose name starts with an underscore. Really, they can reimplement the three lines in `replace` that appear before the call to `_replace`, that's a negligible repetition.